### ISSUE TITLE
Naming/VariableNumber reports an offense for an empty symbol

### ISSUE THREAD

**Issue by pyromaniac** [rubocop/rubocop#15507]
## Before submitting

- [x] I searched the existing issues and this has not been reported.
- [x] I am running the latest stable RuboCop release.

## Issue type

False positive

## Cop name

`Naming/VariableNumber`

## Steps to reproduce the problem

Create `test.rb`:

```ruby
expected = { "": [] }
```

The quoted label is an empty symbol key (`:""`).

Run:

```console
$ bundle exec rubocop --only Naming/VariableNumber test.rb
```

The same predicate is still present on the current `master` branch. The existing spec covers integer symbols, but not an empty symbol.

## Expected behavior

No offense is registered. The empty symbol contains no number, so none of the cop's numbering styles apply.

## Actual behavior

```text
== test.rb ==
C:  1: 14: Naming/VariableNumber: Use normalcase for symbol numbers.

1 file inspected, 1 offense detected
```

The empty name falls through the configured numbering regex because each supported format requires at least one character.

## RuboCop configuration

Default `Naming/VariableNumber` configuration (`EnforcedStyle: normalcase`, `CheckSymbols: true`).

## RuboCop version

```text
1.88.2 (using Parser 3.3.12.0, Prism 1.9.0, rubocop-ast 1.50.0, analyzing as Ruby 3.3, running on ruby 3.4.9) [arm64-darwin25]
```
